### PR TITLE
PIM-3730: Fix variant group link on product edit page

### PR DESCRIPTION
**Pull request by solivier**
| Q                    | A
| -------------------- | ---
| Bug fix?             | Yes
| New feature?         | No
| BC breaks?           | Yes
| CI currently passes? | Yes
| Tests pass?          | Yes
| Scenarios pass?      | Running
| Checkstyle issues?*  | No
| PMD issues?**        | No
| Changelog updated?   | Yes
| Fixed tickets        | PIM-3730
| DB schema updated?   | No
| Migration script?    | No
| Doc PR               | -

### CODE REVIEW

File: src/Pim/Bundle/CatalogBundle/Repository/ProductRepositoryInterface.php [hunk 1]
```diff
@@ -143,4 +143,12 @@ public function findOneById($id);
...
      */
     public function getEligibleProductIdsForVariantGroup($variantGroupId);
+
+    /**
+     * @param GroupInterface $group
+     * @param                $maxResults
+     *
+     * @return array
+     */
+    public function getProductsByGroup(GroupInterface $group, $maxResults);
```
Comment (nidup) [inline src/Pim/Bundle/CatalogBundle/Repository/ProductRepositoryInterface.php]: add a method in an interface is a BC Break Bro

File: src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/ProductRepository.php [hunk 2]
```diff
@@ -710,4 +710,31 @@ public function findByReference($code)
...
+            array_pop($products);
+            $qb = $this->createQueryBuilder('p')
+                ->hydrate(false)
+                ->field('groupIds')->in([$group->getId()]);
+
+            $count = $qb->getQuery()->execute()->count();
+        }
+
+        return [
+            'products'      => $products,
```
Comment (Nuscly) [inline src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/ProductRepository.php]: Possible memory problem if there are a lot of products

File: src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/ProductRepository.php [hunk 3]
```diff
@@ -710,4 +710,32 @@ public function findByReference($code)
...
+
+        return $products;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getProductCountByGroup(GroupInterface $group)
+    {
+            $qb = $this->createQueryBuilder('p')
```
Comment (nidup) [inline src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/ProductRepository.php]: alignement

File: src/Pim/Bundle/CatalogBundle/Manager/GroupManager.php [hunk 4]
```diff
@@ -177,33 +182,10 @@ public function remove($group, array $options = [])
      */
     public function getProductList(GroupInterface $group, $maxResults)
```
Comment (nidup) [inline src/Pim/Bundle/CatalogBundle/Manager/GroupManager.php]: you can spec this method

**Pull request merged**